src/components/KTreeList/KTreeList.vue

```diff
@@ -0,0 +1,419 @@
+  }
+
+  const sumLevel = levelOfDropLocation + deepestLevel
+  if (sumLevel > props.maxLevel) {
+    // disallow movement
+    return false
+  }
+
+  // allow movement
+  return true
+}
+
+const onStartDrag = (): void => {
+  dragging.value = true
+  setDragCursor(true)
+}
+
+const onStopDrag = (target: any): void => {
+  dragging.value = false
+  setDragCursor(false)
+
+  const item = target.draggedContext?.element
+  const parent = target.relatedContext?.element
+  if (item && parent) {
+    handleSetParent(item, parent)
+  }
+  key.value++
+}
+
+// override cursor when dragging
+const setDragCursor = (value: boolean) => {
+  // must be on html element to keep style applied no matter where they drag
+  const html = document.getElementsByTagName('html').item(0)
+
+  if (html) {
+    html.classList.toggle('k-tree-list-grabbing', value)
+  }
+}
+
+// watch for programmatic changes to modelValue
+watch(value, (newVal, oldVal) => {
+  if (JSON.stringify(newVal) !== JSON.stringify(oldVal)) {
+    internalList.value = newVal
+  }
+})
+
+watch(() => props.items, (newValue, oldValue) => {
+  // Only trigger the watcher if items actually change
+  if (JSON.stringify(newValue) === JSON.stringify(oldValue)) {
+    internalList.value = newValue
+  }
+})
+
+onMounted(() => {
+  if (props.items) {
+    internalList.value = props.items
+  } else if (props.modelValue) {
+    internalList.value = props.modelValue
+  }
+
+  internalList.value.forEach((item: TreeListItem) => {
+    if (!item.children) {
+      item.children = []
+    }
+  })
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
+@import '@/styles/functions';
+
+.k-tree-list {
+  .child-drop-zone {
+    // this is the height of the area you can drop an item in
+    // to make it the child of another item
+    min-height: 4px;
+  }
+
+  .has-children {
+    margin-bottom: 4px;
+  }
+
+  // style while dragging an item
+ .k-tree-item-dragged {
+    border-bottom: 4px solid var(--teal-200);
+  }
+
+  .k-tree-item-container.non-root {
+    $border: var(--grey-200);
+    $indent: 8px;
+    $left: -($indent);
+    position: relative;
+
+    &:before {
+      content: "";
+      position: absolute;
+      top: -5px;
+      left: $left;
+      border-left: 1px solid $border;
+      border-bottom: 1px solid $border;
+      border-radius: 0 0 0 5px;
+      width: $indent;
+      height: 20px;
+    }
+    &:after {
+      position: absolute;
+      content: "";
+      top: 12px;
+      left: $left;
+      border-left: 1px solid $border;
+      width: $indent;
+      height: 100%;
+    }
+    &:last-child:after {
+      display: none;
+    }
+  }
+}
+</style>
+
+<style lang="scss">
+// override cursor as grabbing when an item is being dragged
+.k-tree-list-grabbing * {
+  cursor: move !important; /* fallback: no `url()` support or images disabled */
+  cursor: -webkit-grabbing !important; /* Chrome 1-21, Safari 4+ */
+  cursor: -moz-grabbing !important; /* Firefox 1.5-26 */
```

Comment:
You shouldn't need the vendor prefixes here: https://caniuse.com/?search=grabbing
```suggestion
```